query-service-impl/src/test/java/org/hypertrace/core/query/service/QueryRequestBuilderUtils.java

```diff
@@ -120,6 +120,10 @@ public static Filter createInFilter(String column, List<String> values) {
     return createFilter(column, Operator.IN, createStringArrayLiteralValueExpression(values));
   }
 
+  public static Filter createBooleanInFilter(String column, List<Boolean> values) {
+    return createFilter(column, Operator.IN, createBooleanArrayLiteralValueExpression(values));
```

Comment:
Basic question: what does it even mean to have a filter where rhs is a list of boolean values? Given that boolean can have only 2 valid values, expected value should be exactly one of them. 
- If both were to be passed together (e.g. `IN(true, false)`), that effectively means boolean value doesn't matter (and hence don't even pass the filter).
- If same value is passed multiple times (e.g. `IN(true, true, true)`), that is equivalent to just `EQ(true)`.

Reply:
I think it's mostly just a consistency question - because yes, you're right it doesn't unlock any functionality that isn't there with either 0 or 1 other filters (unless it's a nullable boolean - which we do actually have). But either way by allowing it, we allow certain logic paths to not have to special case boolean values and translate different attribute type filters differently.

Reply:
Not sure about nullable boolean, but in both cases above it could help optimise the filters. In first case, we can just skip passing the filter. In second case, we can use EQ which is faster than IN.

Reply:
Yeah - no issues if we want to translate them at this layer - I just wouldn't force the optimization on the caller.

Reply:
If a simple boolean type suffice, there may not be a need for an API interface with a BOOLEAN_ARRAY operator. But, I guess it was added for convenient for nullable boolean having three buckets (true, false, null).

In that case, do we expect the query like below:

```
childFilter {
  childFilter {
    lhs {
      attribute_expression {
        attributeId: "EVENT.isApiExternal"
        alias: "isApiExternal"
      }
    }
    operator: IN
    rhs {
      literal {
        value {
          valueType: BOOLEAN_ARRAY
          boolean_array: null
          boolean_array: true
        }
      }
    }
  }
}
```
If yes, we would need to ensure that we store all boolean fields as string fields in Pinot. We would need to enforce this during pinot schema creation for a view. And we have to go with `value as string` approach so PR #170 

If we store them as boolean type in pinot, nullable boolean will be consider as boolean value `false`. That will loose `null` bucket.

Regarding performance, there won't be much difference if data type is boolean in Pinot. For boolean data type, using IN (true, true, true) is same as EQ(true). And for IN(true, false), the suggestion will make difference if the column is exclusively used in filtering. We can make these change. But, looks like we will have to go with `string` base approach to include `null` bucket.

Reply:
> If yes, we would need to ensure that we store all boolean fields as string fields in Pinot. 

What's our state today? If we are mixed, then we need to include null support including IN clauses. If we're consistently boolean, seems like we can make a few optimizations like dropping the in(true, false) case. I wouldn't build to support the hypothetical nullable boolean if we're not using it.

Reply:
>What's our state today? 

Our current system has a mixed approach for pinot schema regarding Avro boolean data types. Except one view,  Avro booleans are stored as strings in pinot. This change in schema behavior is a result of upgrade of Pinot client library. Previously, the it treated booleans as strings, but now it treats them as booleans.

The pinot query works as string input for boolean field. Below are valid queries:
```
-- value as string
select api_is_external, api_id from bareSpanEventView
where start_time_millis > 1693470600000 and start_time_millis < 1693474200000
and is_external IN ('true')
limit 10
```
or
```
-- value as string
select api_is_external, api_id from bareSpanEventView
where start_time_millis > 1693470600000 and start_time_millis < 1693474200000
and is_external IN ('null')
limit 10
```

We are already treating boolean as string in [QS](https://github.com/hypertrace/query-service/blob/559f23bdbc90e5ab71af268a55d95b932231ffc7/query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java#L499). So, applying same for boolean_array.